python/cudf/cudf/tests/test_list.py

```diff
@@ -112,3 +112,32 @@ def test_len(data):
     got = gsr.list.len()
 
     assert_eq(expect, got, check_dtype=False)
+
+
+@pytest.mark.parametrize(
```

Comment:
May need more test cases to cover:
https://github.com/rapidsai/cudf/blob/c1c60ba3daf36d0ee5553558f70669f454d9f0c8/cpp/include/cudf/lists/contains.hpp#L37-L41

Reply:
For 1 and 2, what is the correct dtype for the null `search_key` when constructing the Scalar? 
i.e  ` search_key= cudf.Scalar(value=None, dtype=?) .`

Reply:
Sorry about late reply. I think for libcudf even to work, they have to be:
1. Of the same type of your list column elements https://github.com/rapidsai/cudf/blob/a5684328726bfa8dfdbbf10a50f47677ce57974b/cpp/src/lists/contains.cu#L147-L148
2. Is one these types https://github.com/rapidsai/cudf/blob/a5684328726bfa8dfdbbf10a50f47677ce57974b/cpp/src/lists/contains.cu#L57-L58

But then another question to ask is, what error should be raised if the user passes in different types of scalars?

Reply:
Reading @mythrocks 's PR thread, it looks like these behaviors are to align with SQL's `array_contain`. Should cudf follow the same semantics? @kkraus14

Reply:
I think requiring the types to match in the Cython function is absolutely reasonable. If we want to handle automatically typecasting we can do so from the Python and do it in a follow up.

I figure a `TypeError` would be most appropriate to raise here, but since libcudf already does it for us, we could do so via a `try / except`?

Reply:
Hmm, I would like to discuss about the use of `try/except` here. My argument against it is that checking the datatype of a column is almost trivial, as compared to one like `null_counts`. So a double check is acceptable here. The downside of `try/except` is that python starts to depend on the error strings that libcudf throws - which is rather a random match at this stage and libcudf has no guarantee of the actual string being thrown (besides `cudf::logic_error` type).

So I suggest we do explicit python check if possible and resort to try/except when the check is non-trivial. I'm open to suggestions. cc @shwina

Reply:
try...except would be Pythonic if we could catch anything more specific than a generic RuntimeError. It seems like the only mechanism libcudf has to communicate any more information than "something went wrong" is the error message itself. On the other hand, relying on string matching seems fragile, especially given that error messages aren't really guaranteed to remain the same.

Reply:
(Dup of PM) It would be great if libcudf has a set error classes that inherits `std::runtime_error`, which can be captured by Cython if thrown. These error classes can maintain some common error interfaces like wrong types; out of bounds access; unsupported operation etc.